magefiles/utils.go

```diff
@@ -441,8 +441,13 @@ func HandleErrorWithAlert(err error, errLevel slack.ErrorSeverityLevel) error {
 	return nil
 }
 
-func getChangedFiles() (rulesengine.Files, error) {
+func getChangedFiles(repo string) (rulesengine.Files, error) {
 	var gitDiff = sh.OutCmd("git", "diff")
+	// If the repo is infra-deployments then we check changed files in path tmp/infra-deployments
+	// else it will run locally which is from e2e-tests directory
+	if repo == "infra-deployments" {
+		gitDiff = sh.OutCmd(fmt.Sprintf("git -C tmp/%s", repo), "diff")
```

Comment:
did you test this? I ran it locally and hit this error
```
exec: git -C tmp/infra-deployments "diff" "--name-status" "main..HEAD"
E0814 16:47:48.715244   82728 utils.go:455] Failed to run git status locally
Error: failed to run "git -C tmp/infra-deployments diff --name-status main..HEAD: fork/exec git -C tmp/infra-deployments: no such file or directory"
```

I think you need to specify the absolute path

Reply:
yes, 
ran it locally on both repos  an it worked for me , 
how did you run it ?

Reply:
updated the `RunRuleDemo` mage target and ran

```
➜  e2e-tests git:(KONFLUX-3187) ✗ ./mage -v local:runruledemo    
Running target: Local:RunRuleDemo
exec: git -C tmp/infra-deployments "diff" "--name-status" "main..HEAD"
E0814 17:17:55.462549   84437 utils.go:454] Failed to run git status locally
Error: failed to run "git -C tmp/infra-deployments diff --name-status main..HEAD: fork/exec git -C tmp/infra-deployments: no such file or directory"
```

Reply:
also when I fixed it, not rules were applied

```
I0814 17:28:47.970449   84868 utils.go:473] The following files, components/build-service/base/kustomization.yaml, were changed!
I0814 17:28:47.970590   84868 types.go:72] Loading the catalog for, infra-deployments, from category, tests
I0814 17:28:47.970595   84868 types.go:144] The following rules have matched .
I0814 17:28:47.970597   84868 types.go:169] Will apply rules
```

Reply:
Hmm, I think what has to be passed into the `getChangedFiles` is the path generated by the installer in order for this to work properly. Especially in CI 

https://github.com/konflux-ci/e2e-tests/blob/59112686eb046d4e8bae4b979fc6e0348150e03c/magefiles/installation/install.go#L107

Not this 

https://github.com/konflux-ci/e2e-tests/pull/1323/files#diff-d7a1577b2d546e34aaf816bdd1b83edef28a66a555f049767bafa83268b7efbbR350

And this if check should just check if the parameter/arg for path is empty. If it is empty, default to the existing way we do the git diff. If it is not, create the git diff using the -c <path arg>. Essentially, lets not force this function to build the path for us. Lets build the path outside of the function

Local dev testing I think that infra-deployments repo should be cloned into that same directory like we do for CI and again we just pass the path to the function. 

Thoughts?

Reply:
>Hmm, I think what has to be passed into the getChangedFiles is the path generated by the installer in order for this to work properly

but we'd need to init the installation controller in order to get the path, which I think is an overkill. Other option could be to put that variable to `pkg/consts.go`, so it will be accessible from both install controller and rule engine, wdyt?

> Local dev testing I think that infra-deployments repo should be cloned into that same directory like we do for CI

we actually do clone the infra-deployments to `e2e-tests/tmp/infra-deployments` in CI

Reply:
> I agree but we'd need to init the installation controller in order to get the path, which I think is an overkill.
> 
> Other option could be to put that variable to magefiles/constants/constants.go or pkg/constants/constants.go, so it will be accessible from both install controller and rule engine, wdyt?

I agree, that is what I was thinking as well. That it be some global const that gets used in both context. 

> we actually do clone the infra-deployments to e2e-tests/tmp/infra-deployments in CI

Right, I was just commenting for local evaluating of rules for infra-deployments we would need to have that documented as pre-req. That if you need to clone it into that path

Reply:
I've fixed it so now the diff will look on the absolute path 
ran it locally and it works fine . 
WDYT?